### DISCUSSION TITLE
is it possible to natively configure a single DLT topic for all failures?

### DISCUSSION THREAD

**Discussion by Watlas** [spring-projects/spring-kafka#2238] [Q&A] [answered]
I'm using @RetryableTopic to configure retries in their respective queues, but when it falls into DLT I want it sent to a specific topic and only to it, is it possible to configure this natively with spring kafka?

Example:

![image](https://user-images.githubusercontent.com/57519850/164247729-00175de7-6b92-41eb-a1be-32c0a7cef3e7.png)

**Comment by tomazfernandes** [comment, marked as answer]
Hi @Watlas, thanks for bringing this up. This is not a feature the framework currently offers, but there's a workaround that might achieve the result you're looking for.

You can provide a custom `RetryTopicNamesProviderFactory` and override the `getTopicName` method when the topic is a `DLT`., such as:

```java
@Bean
public RetryTopicNamesProviderFactory providerFactory() {
	return new SuffixingRetryTopicNamesProviderFactory() {
		@Override
		public RetryTopicNamesProvider createRetryTopicNamesProvider(DestinationTopic.Properties properties) {
			if (properties.isDltTopic()) {
				return new SuffixingRetryTopicNamesProvider(properties) {
					@Override
					public String getTopicName(String topic) {
						return "my.dlt";
					}
				};
			}
			return super.createRetryTopicNamesProvider(properties);
		}
	};
}
```

Mind that you'll still get one consumer per retryable topic for the dlt. You might have only one `@DltHandler`, and then the other consumers will use the default logging handler. Since they'll belong to different consumer groups, they shouldn't interfere one with the other.

I think this is an interesting feature request, if you'd like to open an issue for this I think we can look into having an official, more optimal way of handling this.

Just as a note, I see you currently have a few `FATAL` exceptions in the topic. Since `2.8.3`, there's a [global set of `FATAL` exceptions](https://docs.spring.io/spring-kafka/reference/html/#retry-topic-ex-classifier) you can manage.

**Reply by Watlas** [reply]
Thank you very much, it worked perfectly.

**Reply by tomazfernandes** [reply]
Glad to hear that, and thanks again for bringing this up.
